### DISCUSSION TITLE
Possibility of text extraction using coordinates

### DISCUSSION THREAD

**Discussion by sandeepreddy5** [jsvine/pdfplumber#831] [Q&A]
## Describe the bug

How would i extract text using coordinates or along with coordinates?

## Code to reproduce the problem

with pdfplumber.open("example.pdf"
[example.pdf](https://github.com/jsvine/pdfplumber/files/10882687/example.pdf)
) as pdf:
    for page in pdf.pages:
        print(page.rects)

## PDF file

example.pdf

## Expected behavior

{'x0': 0.0, 'y0': 0.0, 'x1': 612.0, 'y1': 792.0, 'width': 612.0, 'height': 792.0, 'pts': [(0.0, 0.0), (612.0, 0.0), (612.0, 792.0), (0.0, 792.0)], 'linewidth': 0, 'stroke': False, 'fill': True, 'evenodd': False, 'stroking_color': (1, 1, 1), 'non_stroking_color': (1, 1, 1), 'object_type': 'rect', 'page_number': 1, 'top': 0.0, 'bottom': 792.0, 'doctop': 0.0, **'<text': 'respective text'**}

## Actual behavior

{'x0': 0.0, 'y0': 0.0, 'x1': 612.0, 'y1': 792.0, 'width': 612.0, 'height': 792.0, 'pts': [(0.0, 0.0), (612.0, 0.0), (612.0, 792.0), (0.0, 792.0)], 'linewidth': 0, 'stroke': False, 'fill': True, 'evenodd': False, 'stroking_color': (1, 1, 1), 'non_stroking_color': (1, 1, 1), 'object_type': 'rect', 'page_number': 1, 'top': 0.0, 'bottom': 792.0, 'doctop': 0.0}

## Environment

- pdfplumber version: [e.g., 0.5.22]
- Python version: [e.g., 3.8.1]
- OS: [e.g., Mac, Linux, etc.]

**Comment by jsvine** [comment]
Hi @sandeepreddy5; you can do this:

```python
with pdfplumber.open("example.pdf") as pdf:
  for page in pdf.pages:
    for rect in page.rects:
        bbox = pdfplumber.utils.obj_to_bbox(rect)
        print(page.crop(bbox).extract_text())
```

**Reply by sandeepreddy5** [reply]
Thanks @jsvine for timely response, but with the above code text and coordinates for the second column is completely missing, only left side part of the text and coordinates are getting printed.
I'm providing the screenshot of the text which is missing.

![Screenshot from 2023-03-06 11-20-08](https://user-images.githubusercontent.com/73211802/223029256-f1a3e279-3494-4050-955c-b5614a08b13a.png)

**Reply by samkit-jain** [reply]
This is because that column is not wrapped in a `rect` object.
